src/main/java/hlf/java/rest/client/service/impl/EventFulfillmentImpl.java

```diff
@@ -0,0 +1,108 @@
+package hlf.java.rest.client.service.impl;
+
+import hlf.java.rest.client.config.FabricProperties;
+import hlf.java.rest.client.exception.ErrorCode;
+import hlf.java.rest.client.exception.ErrorConstants;
+import hlf.java.rest.client.exception.NotFoundException;
+import hlf.java.rest.client.exception.ServiceException;
+import hlf.java.rest.client.listener.BlockEventListener;
+import hlf.java.rest.client.listener.ChaincodeEventListener;
+import hlf.java.rest.client.model.ClientResponseModel;
+import hlf.java.rest.client.model.EventType;
+import hlf.java.rest.client.service.EventFulfillment;
+import java.util.Objects;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.hyperledger.fabric.gateway.Gateway;
+import org.hyperledger.fabric.gateway.Network;
+import org.hyperledger.fabric.sdk.BlockEvent;
+import org.hyperledger.fabric.sdk.BlockInfo;
+import org.hyperledger.fabric.sdk.ChaincodeEvent;
+import org.hyperledger.fabric.sdk.Channel;
+import org.hyperledger.fabric.sdk.exception.InvalidArgumentException;
+import org.hyperledger.fabric.sdk.exception.ProposalException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.context.config.annotation.RefreshScope;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@ConditionalOnProperty(prefix = "fabric.events", name = "enable", havingValue = "true")
+@RefreshScope
+public class EventFulfillmentImpl implements EventFulfillment {
+
+  @Autowired private FabricProperties fabricProperties;
+
+  @Autowired private Gateway gateway;
+
+  @Autowired private BlockEventListener blockEventListener;
+
+  @Autowired private ChaincodeEventListener chaincodeEventListener;
+
+  /**
+   * This function replays all the events from blockchain It returns the type of event requested
+   * i.e. BLOCK_EVENT and CHAINCODE_EVENT
+   *
+   * @param startBlockNumber block number since when the events are requested
+   * @param endBlockNumber block number until when to replay events
+   * @param networkName channel name
+   * @param eventType type of event
+   * @return the ClientResponseModel with the status that request is honored.
+   */
+  @Override
+  public ResponseEntity<ClientResponseModel> replayEvents(
```

Comment:
TODO: These can be picked up later as an improvement.
1. Graceful error handling and the status corresponding to that.
2. Asynchronous processing of replay.
3. Make the end block number optional.